pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -556,6 +556,34 @@ func (p *Pod) groupTotalCount() (int, error) {
 	return gtc, nil
 }
 
+// podGroupIndex returns the value of GroupIndexLabel for the pod being reconciled at the moment.
+func (p *Pod) podGroupIndex() (*int, error) {
+	if podGroupName(p.pod) == "" {
+		return nil, fmt.Errorf("pod doesn't have a '%s' label", GroupNameLabel)
+	}
+	podGroupTotalCount, err := p.groupTotalCount()
```

Comment:
This is something we could pass down from the context to avoid parsing again

Reply:
Fixed